files/default/sql/ddl/updates/3.7.0.sql

```diff
@@ -101,25 +101,199 @@ CREATE TABLE IF NOT EXISTS `hopsworks`.`feature_group_descriptive_statistics` (
+  FROM `hopsworks`.`feature_store_statistic` WHERE id BETWEEN start_id AND end_id AND id IN (SELECT id FROM `hopsworks`.`feature_group_statistics`);
+
+  -- -- insert one feature descriptive statistic per orphan feature group statistics file, to be deleted during the expat
+  INSERT INTO `hopsworks`.`feature_descriptive_statistics` (id, feature_name, feature_type, count, extended_statistics_path)
   SELECT id, 'to-be-deleted', 'FEATURE_GROUP' as feature_type, feature_group_id, file_path
-  FROM `feature_store_statistic` WHERE feature_group_id IS NOT NULL AND id NOT IN (SELECT id FROM `feature_group_statistics`);
--- To be done in the expat: insert feature_group_descriptive_statistics (many-to-many relationship), parse feature_descriptive_statistics, delete orphan fg statistics files
+  FROM `hopsworks`.`feature_store_statistic` WHERE feature_group_id IS NOT NULL AND id BETWEEN start_id AND end_id AND id NOT IN (SELECT id FROM `hopsworks`.`feature_group_statistics`);
+  -- To be done in the expat: insert feature_group_descriptive_statistics (many-to-many relationship), parse feature_descriptive_statistics, delete orphan fg statistics files
+END //
+
+CREATE PROCEDURE BatchPopulatePopulateFeatureDescriptiveStatistics()
+BEGIN
+    DECLARE start_id INT;
+    DECLARE end_id INT;
+
+    SELECT MIN(id), MAX(id) INTO start_id, end_id FROM `hopsworks`.`feature_store_statistic`;
+
+    CALL BatchProc('PopulateFeatureDescriptiveStatistics', 1000, start_id, end_id);
+    
+END //
+
+-- Procedure to populate training_dataset_statistics
+CREATE PROCEDURE PopulateTrainingDatasetStatistics(start_id INT, end_id INT)
+BEGIN
+  -- -- insert new training_dataset_statistics. For the same training dataset, only insert the last computed statistics.
+  INSERT INTO `hopsworks`.`training_dataset_statistics` (id, computation_time, training_dataset_id, before_transformation)
+  WITH sorted_feature_store_statistic AS (
+    SELECT `hopsworks`.`feature_store_statistic`.*, ROW_NUMBER() OVER (PARTITION BY training_dataset_id, for_transformation ORDER BY commit_time DESC) as rn -- group by td_id and before_transformation, and sort by commit_time --
+    FROM `hopsworks`.`feature_store_statistic` WHERE training_dataset_id IS NOT NULL AND training_dataset_id BETWEEN start_id AND end_id
+  )
+  SELECT id, commit_time as computation_time, training_dataset_id, for_transformation
+  FROM sorted_feature_store_statistic WHERE rn = 1; -- last computed stats --
+END //
+
+CREATE PROCEDURE BatchPopulateTrainingDatasetStatistics()
+BEGIN
+    DECLARE start_id INT;
+    DECLARE end_id INT;
+
+    SELECT MIN(training_dataset_id), MAX(training_dataset_id) INTO start_id, end_id FROM `hopsworks`.`feature_store_statistic`;
+
+    CALL BatchProc('PopulateTrainingDatasetStatistics', 1000, start_id, end_id);
+    
+END //
+
+-- Procedure to populate feature_descriptive_statistics
+CREATE PROCEDURE PopulateFeatureDescriptiveStatisticsForTraingingDataset(start_id INT, end_id INT)
```

Comment:
There is a type here `PopulateFeatureDescriptiveStatisticsForTraingingDataset` (Trainging)